notifications/templatetags/notifications_tags.py

```diff
@@ -68,7 +70,10 @@ def register_notify_callbacks(badge_class='live_notify_badge',  # pylint: disabl
         fetch_count=fetch
     )
 
-    script = "<script>" + definitions
+    # add a nonce value to the script tag if one is provided
+    nonce_str = ' nonce="{nonce}"'.format(nonce=nonce) if nonce is not None else ""
```

Comment:
If the variable doesn't exist in the template context, the default value is an empty string, so checking this against `None` will be `true` and the nonce attribute will be filled with an empty string.

```suggestion
    nonce_str = ' nonce="{nonce}"'.format(nonce=nonce) if nonce else ""
```